dowhy/graph.py

```diff
@@ -187,13 +188,54 @@ def is_blocked(graph: nx.DiGraph, path, conditioned_nodes):
         return False
 
 
+def get_ancestors(graph: nx.DiGraph, nodes):
```

Comment:
I noticed that get_descendants (line 198) does not include the nodes themselves, so I've replicated that here. Though when we invoke this method below we will have to union the result with the nodes themselves.